crates/networking/p2p/types.rs

```diff
@@ -284,38 +276,42 @@ pub struct NodeRecordPairs {
     pub secp256k1: Option<H264>,
     // https://github.com/ethereum/devp2p/blob/master/enr-entries/eth.md
     pub eth: Option<ForkId>,
+    // Snap entry is being used by some tests such as `test_encode_enr_response`.
+    pub snap: Option<Vec<u32>>,
     // TODO implement ipv6 specific ports
 }
 
-impl NodeRecord {
-    pub fn decode_pairs(&self) -> NodeRecordPairs {
+impl NodeRecordPairs {
+    pub fn try_from_raw_pairs(
+        pairs: &Vec<(Bytes, Bytes)>,
+    ) -> Result<NodeRecordPairs, RLPDecodeError> {
         let mut decoded_pairs = NodeRecordPairs::default();
-        for (key, value) in &self.pairs {
+        for (key, value) in pairs {
             let Ok(key) = String::from_utf8(key.to_vec()) else {
```

Comment:
Although the same probably can be said about this. If it's not valid UTF8 should we still take the packet?